packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1391,22 +1391,60 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction hash associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param hash - The new hash value to be assigned.
+   */
+  updateTransactionHash(transactionId: string, hash: string) {
+    const transactionMeta = this.getTransaction(transactionId);
+    if (!transactionMeta) {
+      log(`Transaction with id ${transactionId} not found.`);
```

Comment:
If we're doing something as important as updating the hash or status, should we instead throw here as we do in the other update methods?

It could avoid error flows being ignored and ensure more stability?

Reply:
Make sense, I've included also to thorow when it's not a custodian.